File: 217.contains-duplicate.js
/*
 * @lc app=leetcode id=217 lang=javascript
 *
 * [217] Contains Duplicate
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function (nums) {
  const lookup = new Set();
  for (const num of nums) {
    if (lookup.has(num)) {
      return true;
    }
    lookup.add(num);
  }
  return false;
};
// @lc code=end

File: 217.contains-duplicate.go
/*
 * @lc app=leetcode id=217 lang=golang
 *
 * [217] Contains Duplicate
 */

package leetcode

// @lc code=start
func containsDuplicate(nums []int) bool {
	lookup := map[int]bool{}
	for i := 0; i < len(nums); i++ {
		if _, ok := lookup[nums[i]]; ok {
			return true
		}
		lookup[nums[i]] = true
	}
	return false
}

// @lc code=end
